### ISSUE TITLE
Can base model be larger than target model?

### ISSUE THREAD

**Issue by jhj0411jhj** [microsoft/mup#39]
This might be a naive question. In the provided examples, it seems that all base models are small models and target models are larger ones. However, if I have a large model to be tuned, and I don't want to change the original training process of the large model, can I set the large model as the base model? In my understanding, `mup` behaves the same as torch when `model.width == base_model.width`.

To illustrate:
```python
# suppose I have a large model: large_model = Model(width=100)
# build mup model
base_model = mupModel(width=100)
smaller_model = mupModel(width=10)

# train and tune lr in the smaller model
best_lr = tune(smaller_model)

# train large model with best lr once
# Does Model(width=100) == mupModel(width=100) ?
train(Model(width=100), best_lr)
```
Is the above example correct?

**Comment by thegregyang** [comment]
There is no problem with this except for the quality of HP transfer.

If you were to tune all hyperparameters (LR and init for every parameter tensor), then this is fine. However, in practice you probably don’t do this, and the things you do not tune are fixed to the default values of the large model. Typically speaking, the default values of large models are less trustworthy than the scaled-up versions of the default values of the small models. For example, if you are only tuning a global learning rate then I would suspect the quality of the large model will be far from the optimal, because you are missing the potential gains from per layer LR and init, which can be dramatically different for different layers, especially in large models (this is what muP predicts). 

Also see [this comment](https://github.com/microsoft/mup/issues/28#issuecomment-1310506926). In particular,

> The true hyperparameter space here is the very high dimensional space containing [learning rate, initialization] (we can insert multipliers here as well, but like I said, it is redundant) for every parameter tensor (weights, biases, gains, etc). If you were to tune all these hyperparameters and obtain the optimal combination, then this combination is guaranteed to be stable in some sense as you vary width (in muP). However, in practice, we may not want to tune that many hyperparameters because of resource constraints. So we combine hyperparameters (by e.g., tying learning rate for many weights together) until we have only a small number to tune. This essentially means that we are now focusing on a low dimensional slice of the true hyperparameter space --- that we guess should contain all the really good hyperparameters. The choices of hyperparameters we tuned in our paper exemplify the “low dimensional slice” we chose. These choices are based on our empirical experience tuning hyperparameters, but over time people may find better choices.

**Comment by jhj0411jhj** [comment]
@thegregyang Thanks! Sometimes we don't want to make modifications to existing large models in production for stability issues, but the thought of tuning per layer LR and init in muP is inspirational, I'll give it a try.

**Comment by thegregyang** [comment]
OK sounds good. I'll close this for now but feel free to reopen if there are further issues.
